File: bookcode/kotlinproject/kotlin-code/src/main/java/cn/danao/main/model/kotlin/SpellTest.java
package cn.danao.main.model.kotlin;

import cn.danao.main.model.bean.OneModel;

/**
 * date 2020/11/20 14:56 <br/>
 * description class <br/>
 *
 * @author danao
 * @version 1.0
 * @since 1.0
 */
public class SpellTest {

    public static void main(String[] args) {
        test(null);
        test(new OneModel());
        test(OneModel.init());
    }

    public static void test(OneModel oneModel) {
        StringBuffer stringBuffer = new StringBuffer();
        if (oneModel == null) {
            return;
        }
        if (oneModel.getStrs() != null) {
            oneModel.getStrs().forEach(k -> {
                stringBuffer.append("oneModel中strs").append(k);
            });
        }
        if (oneModel.getTwoModels() != null) {
            oneModel.getTwoModels().forEach(twoModel -> {
                if (twoModel != null) {
                    if (twoModel.getThreeModels() != null) {
                        twoModel.getThreeModels().forEach(threeModel -> {
                            if (threeModel != null) {
                                if (threeModel.getStrs() != null) {
                                    threeModel.getStrs().forEach(s -> {
                                        stringBuffer.append("threeModel中的strs").append(s);
                                    });
                                }
                            }
                        });
                    }
                }
            });
        }
        System.out.println("str = " + stringBuffer);
    }
}

File: bookcode/kotlinproject/kotlin-code/src/main/java/cn/danao/main/model/kotlin/SpellTest.kt
package cn.danao.main.model.kotlin

import cn.danao.main.model.bean.OneModel

fun main() {
    var init = OneModel.init()
    spell(null)
    spell(OneModel())
    spell(init)
}


fun spell(oneModel: OneModel?) {
    var stringBuffer = StringBuffer()
    stringBuffer.append(oneModel?.getId() ?: "").append(oneModel?.name ?: "")
    println("str = $stringBuffer")
    oneModel?.strs?.forEach { stringBuffer.append("oneModel的strs,$it") }
    oneModel?.twoModels?.forEach {
        it?.threeModels?.forEach { three ->
            three?.strs?.forEach { strs ->
                stringBuffer.append("threemodel中的strs$strs")
            }
        }
    }
    println("str = $stringBuffer")
    println("end")
}